File: library/src/main/java/com/thesurix/gesturerecycler/util/FixedSizeArrayDequeue.java
package com.thesurix.gesturerecycler.util;

import java.util.ArrayDeque;

/**
 * @author thesurix
 */
public class FixedSizeArrayDequeue<E> extends ArrayDeque<E> {

    private int mMaxSize;

    public FixedSizeArrayDequeue(final int maxSize) {
        super(maxSize);
        mMaxSize = maxSize;
    }

    @Override
    public boolean offer(final E e) {
        if (size() == mMaxSize) {
            removeFirst();
        }

        return super.offer(e);
    }
}

File: library/src/main/java/com/thesurix/gesturerecycler/util/FixedSizeArrayDequeue.kt
package com.thesurix.gesturerecycler.util

import java.util.*

/**
 * @author thesurix
 */
class FixedSizeArrayDequeue<E>(private val maxSize: Int) : ArrayDeque<E>(maxSize) {

    override fun offer(e: E): Boolean {
        if (size == maxSize) {
            removeFirst()
        }

        return super.offer(e)
    }
}
